storage/isolation_manager.rs

```diff
@@ -440,9 +440,12 @@ impl Timeline {
         SequenceNumber::from(self.watermark.load(Ordering::SeqCst))
     }
 
-    fn record_reader(&self, sequence_number: SequenceNumber) {
+    fn record_reader(self: Arc<Self>, sequence_number: SequenceNumber) -> ReaderDropGuard {
         if let Some(window) = self.try_get_window(sequence_number) {
             window.increment_readers();
+            ReaderDropGuard { timeline: self.clone(), sequence_number }
+        } else {
+            unimplemented_feature!(ReopenOldSnapshotMVCC)
```

Comment:
We now just return a fully self contained drop guard.

Interestingly, this highlighted what would be a bug - opening a txn on an old MVCC snapshot that's not got a window in memory should be an error/needs to be dealt with.